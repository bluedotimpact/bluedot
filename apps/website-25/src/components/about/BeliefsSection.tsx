import { Section, ValueCard } from '@bluedot/ui';

const BeliefsSection = () => {
  return (
    <Section className="beliefs-section" title="Our core beliefs">
      <div className="beliefs-section__container flex gap-8 overflow-y-scroll">
        <ValueCard
          icon="icons/fast.svg"
          title="AGI could arrive soon, and weâ€™re dangerously unprepared"
          description="Human-level AI will likely be built within years, not decades. Our technical, political, and social systems are not prepared to handle it safely, and the default path leads to catastrophic outcomes."
        />
        <ValueCard
          icon="icons/care.svg"
          title="What we do now matters"
          description="The future of AI isn't set in stone. Our choices today will determine whether this technology helps or harms humanity."
        />
        <ValueCard
          icon="icons/solvers.svg"
          title="The right people at key moments rewrite history"
          description="Throughout history, capable individuals in key positions have steered powerful technologies toward better outcomes. By identifying, training, and mobilising exceptional people, we can guide AI toward a safe and beneficial future."
        />
        <ValueCard
          icon="icons/solvers.svg"
          title="We need urgency, wisdom and optimism"
          description="The stakes are vast and time is short, but panic and fatalism do more harm than good. We need rapid, coordinated action informed by evidence and guided by hope."
        />
      </div>
    </Section>
  );
};

export default BeliefsSection;
