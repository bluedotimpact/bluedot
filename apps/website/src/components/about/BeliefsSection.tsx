import { Card, Section } from '@bluedot/ui';
import { SlideList } from '@bluedot/ui/src/SlideList';

const beliefs = [
  {
    imageSrc: '/images/beliefs/agi.png',
    title: 'AGI could arrive soon, and society is dangerously unprepared',
    subtitle: 'Human-level AI will likely be built within years, not decades. Our technical, political, and social systems are not prepared to handle it safely.',
  },
  {
    imageSrc: '/images/beliefs/matters.png',
    title: 'What we do now matters',
    subtitle: 'The future of AI isn\'t set in stone. Our choices today will determine whether this technology helps or harms humanity.',
  },
  {
    imageSrc: '/images/beliefs/key.png',
    title: 'People at key moments rewrite history',
    subtitle: 'Throughout history, people in key positions have steered powerful technologies toward better outcomes. By training and mobilising exceptional people, we can guide AI toward a safe and beneficial future.',
  },
  {
    imageSrc: '/images/beliefs/urgency.png',
    title: 'We need urgency, wisdom and optimism',
    subtitle: 'The stakes are vast and time is short, but panic and fatalism do more harm than good. We need rapid, coordinated action informed by evidence and guided by hope.',
  },
] as const;

const BeliefsSection = () => {
  return (
    <Section title="Our core beliefs" className="beliefs-section">
      <SlideList
        maxItemsPerSlide={4}
        className="beliefs-section__beliefs"
      >
        {beliefs.map((belief) => (
          <Card key={belief.title} {...belief} className="beliefs-section__belief" />
        ))}
      </SlideList>
    </Section>
  );
};

export default BeliefsSection;
