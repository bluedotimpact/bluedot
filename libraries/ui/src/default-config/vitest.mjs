/* eslint-disable import/no-extraneous-dependencies */
import { defineConfig, defaultExclude } from 'vitest/config';
import react from '@vitejs/plugin-react';
import * as dotenv from 'dotenv';

/**
 * @param {import('vitest/config').UserConfig} [config] 
 * @returns {import('vitest/config').UserConfig}
 */
export const withDefaultBlueDotVitestConfig = async (config) => defineConfig({
  ...config,
  plugins: [react(), ...(config?.plugins ?? [])],
  test: {
    environment: 'happy-dom',
    env: dotenv.config({ path: '.env.test' }).parsed,

    ...config?.test,

    exclude: [...defaultExclude, '**/.{turbo,next}/**', ...(config?.test?.exclude ?? [])],
  },
});
