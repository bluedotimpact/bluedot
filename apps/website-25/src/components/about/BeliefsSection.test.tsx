import { render } from '@testing-library/react';
import { describe, expect, test } from 'vitest';
import BeliefsSection from './BeliefsSection';

describe('BeliefsSection', () => {
  test('renders default as expected', () => {
    const { container } = render(<BeliefsSection />);
    expect(container).toMatchSnapshot();
  });
});
