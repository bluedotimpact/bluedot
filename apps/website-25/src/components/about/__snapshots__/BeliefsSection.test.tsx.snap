// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`BeliefsSection > renders default as expected 1`] = `
<div>
  <div
    class="slide-list w-full flex flex-col gap-spacing-y beliefs-section section-body"
  >
    <div
      class="section-heading__title-container flex justify-between items-center gap-space-between"
    >
      <div
        class="section-heading__content flex-1 flex flex-col gap-2"
      >
        <h2
          class="section-heading__title relative"
        >
          Our core beliefs
        </h2>
      </div>
      <div
        class="slide-list__nav--header items-center gap-2 ml-auto mt-auto flex"
      >
        <button
          aria-label="Previous slide"
          class="slide-list__nav-button p-3 border rounded-lg transition-colors border-charcoal-light text-black hover:bg-gray-50 active:bg-gray-100 disabled:opacity-50 disabled:cursor-not-allowed focus:outline-hidden focus:ring-2 focus:ring-bluedot-light"
          disabled=""
          type="button"
        >
          <svg
            class="slide-list__nav-icon size-5"
            fill="none"
            stroke="currentColor"
            viewBox="0 0 24 24"
          >
            <path
              d="M15 19l-7-7 7-7"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
            />
          </svg>
        </button>
        <button
          aria-label="Next slide"
          class="slide-list__nav-button p-3 border rounded-lg transition-colors border-charcoal-light text-black hover:bg-gray-50 active:bg-gray-100 disabled:opacity-50 disabled:cursor-not-allowed focus:outline-hidden focus:ring-2 focus:ring-bluedot-light"
          type="button"
        >
          <svg
            class="slide-list__nav-icon size-5"
            fill="none"
            stroke="currentColor"
            viewBox="0 0 24 24"
          >
            <path
              d="M9 5l7 7-7 7"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
            />
          </svg>
        </button>
      </div>
    </div>
    <div
      class="slide-list__content flex flex-col lg:flex-row gap-space-between"
    >
      <div
        class="slide-list__container relative overflow-hidden flex-1 flex flex-col gap-space-between"
      >
        <div
          class="slide-list__nav--container items-center justify-end gap-2 hidden"
        >
          <button
            aria-label="Previous slide"
            class="slide-list__nav-button p-3 border rounded-lg transition-colors border-charcoal-light text-black hover:bg-gray-50 active:bg-gray-100 disabled:opacity-50 disabled:cursor-not-allowed focus:outline-hidden focus:ring-2 focus:ring-bluedot-light"
            disabled=""
            type="button"
          >
            <svg
              class="slide-list__nav-icon size-5"
              fill="none"
              stroke="currentColor"
              viewBox="0 0 24 24"
            >
              <path
                d="M15 19l-7-7 7-7"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
              />
            </svg>
          </button>
          <button
            aria-label="Next slide"
            class="slide-list__nav-button p-3 border rounded-lg transition-colors border-charcoal-light text-black hover:bg-gray-50 active:bg-gray-100 disabled:opacity-50 disabled:cursor-not-allowed focus:outline-hidden focus:ring-2 focus:ring-bluedot-light"
            type="button"
          >
            <svg
              class="slide-list__nav-icon size-5"
              fill="none"
              stroke="currentColor"
              viewBox="0 0 24 24"
            >
              <path
                d="M9 5l7 7-7 7"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
              />
            </svg>
          </button>
        </div>
        <div
          class="slide-list__gradient-overlay absolute inset-0 pointer-events-none z-10 flex"
        >
          <div
            class="slide-list__gradient-right absolute inset-y-0 right-0 w-8 bg-linear-to-l from-cream-normal to-transparent opacity-50"
          />
        </div>
        <div
          class="slide-list__slides flex overflow-x-scroll scrollbar-hidden transition-transform duration-300 gap-space-between"
        >
          <div
            class="slide-list__slide shrink-0"
            style=""
          >
            <div
              class="card flex flex-col items-start transition-transform duration-200 beliefs-section__belief"
            >
              <div
                class="card__image-container w-full mb-4"
              >
                <img
                  alt="AGI could arrive soon, and we're unprepared"
                  class="card__image w-full max-h-full object-cover rounded-radius-md "
                  src="/images/beliefs/agi.png"
                />
              </div>
              <div
                class="card__content flex flex-col gap-6 w-full flex-1 justify-between"
              >
                <div
                  class="card__text"
                >
                  <p
                    class="card__title subtitle-sm mb-2"
                  >
                    AGI could arrive soon, and we're unprepared
                  </p>
                  <p
                    class="card__subtitle "
                  >
                    Human-level AI will likely be built within years, not decades. Our technical, political, and social systems are not prepared to handle it safely, and the default path leads to catastrophic outcomes.
                  </p>
                </div>
              </div>
            </div>
          </div>
          <div
            class="slide-list__slide shrink-0"
            style=""
          >
            <div
              class="card flex flex-col items-start transition-transform duration-200 beliefs-section__belief"
            >
              <div
                class="card__image-container w-full mb-4"
              >
                <img
                  alt="What we do now matters"
                  class="card__image w-full max-h-full object-cover rounded-radius-md "
                  src="/images/beliefs/matters.png"
                />
              </div>
              <div
                class="card__content flex flex-col gap-6 w-full flex-1 justify-between"
              >
                <div
                  class="card__text"
                >
                  <p
                    class="card__title subtitle-sm mb-2"
                  >
                    What we do now matters
                  </p>
                  <p
                    class="card__subtitle "
                  >
                    The future of AI isn't set in stone. Our choices today will determine whether this technology helps or harms humanity.
                  </p>
                </div>
              </div>
            </div>
          </div>
          <div
            class="slide-list__slide shrink-0"
            style=""
          >
            <div
              class="card flex flex-col items-start transition-transform duration-200 beliefs-section__belief"
            >
              <div
                class="card__image-container w-full mb-4"
              >
                <img
                  alt="The right people at key moments rewrite history"
                  class="card__image w-full max-h-full object-cover rounded-radius-md "
                  src="/images/beliefs/key.png"
                />
              </div>
              <div
                class="card__content flex flex-col gap-6 w-full flex-1 justify-between"
              >
                <div
                  class="card__text"
                >
                  <p
                    class="card__title subtitle-sm mb-2"
                  >
                    The right people at key moments rewrite history
                  </p>
                  <p
                    class="card__subtitle "
                  >
                    Throughout history, capable individuals in key positions have steered powerful technologies toward better outcomes. By identifying, training, and mobilising exceptional people, we can guide AI toward a safe and beneficial future.
                  </p>
                </div>
              </div>
            </div>
          </div>
          <div
            class="slide-list__slide shrink-0"
            style=""
          >
            <div
              class="card flex flex-col items-start transition-transform duration-200 beliefs-section__belief"
            >
              <div
                class="card__image-container w-full mb-4"
              >
                <img
                  alt="We need urgency, wisdom and optimism"
                  class="card__image w-full max-h-full object-cover rounded-radius-md "
                  src="/images/beliefs/urgency.png"
                />
              </div>
              <div
                class="card__content flex flex-col gap-6 w-full flex-1 justify-between"
              >
                <div
                  class="card__text"
                >
                  <p
                    class="card__title subtitle-sm mb-2"
                  >
                    We need urgency, wisdom and optimism
                  </p>
                  <p
                    class="card__subtitle "
                  >
                    The stakes are vast and time is short, but panic and fatalism do more harm than good. We need rapid, coordinated action informed by evidence and guided by hope.
                  </p>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
